cranelift/codegen/src/isa/x64/lower/isle.rs

```diff
@@ -133,6 +133,62 @@ impl Context for IsleContext<'_, '_, MInst, Flags, IsaFlags, 6> {
         RegMemImm::reg(self.put_in_reg(val))
     }
 
+    fn put_in_xmm_mem_imm(&mut self, val: Value) -> XmmMemImm {
+        let inputs = self.lower_ctx.get_value_as_source_or_const(val);
+
+        if let Some(c) = inputs.constant {
+            if let Some(imm) = to_simm32(c as i64) {
+                return XmmMemImm::new(imm.to_reg_mem_imm()).unwrap();
+            }
+
+            // A load from the constant pool is better than a rematerialization into a register,
```

Comment:
Can we delegate to `put_in_xmm_mem` at this point to share logic (then lift the `XmmMem` back to `XmmMemImm`)?